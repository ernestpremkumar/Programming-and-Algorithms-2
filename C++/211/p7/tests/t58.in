echo 2
breadth
find 2
insert 0
print
breadth
print
size
distance
breadth
distance
breadth
print
print
find 3
distance
distance
breadth
print
breadth
echo 0
echo 2
insert 4
print
distance
find 1
size
find 0
find 6
size
echo 6
insert 0
find 7
echo 7
breadth
balanced
balanced
size
size
insert 6
echo 1
breadth
breadth
insert 0
balanced
distance
find 0
balanced
distance
distance
breadth
size
breadth
breadth
find 0
insert 6
find 3
breadth
balanced
echo 6
insert 6
distance
breadth
find 5
breadth
echo 1
distance
distance
distance
print
size
print
find 5
breadth
echo 2
insert 5
find 2
distance
breadth
breadth
size
print
balanced
breadth
distance
balanced
distance
distance
breadth
print
insert 6
echo 5
print
distance
echo 1
find 3
echo 2
find 7
size
print
insert 0
insert 3
balanced
insert 0
print
size
size
size
insert 6
echo 0
find 1
find 3
find 3
size
balanced
breadth
size
echo 7
balanced
distance
distance
echo 0
find 0
balanced
find 4
breadth
size
size
balanced
size
print
echo 0
balanced
echo 0
size
insert 6
distance
insert 1
print
breadth
balanced
balanced
find 1
balanced
distance
insert 1
insert 2
insert 1
size
distance
balanced
print
insert 5
balanced
breadth
breadth
balanced
distance
distance
print
distance
balanced
balanced
echo 0
insert 1
breadth
distance
breadth
balanced
breadth
insert 5
insert 2
balanced
size
print
insert 6
print
find 0
echo 4
breadth
size
insert 3
breadth
echo 7
find 4
insert 5
echo 4
print
size
print
breadth
size
distance
breadth
echo 6
echo 3
distance
print
echo 3
balanced
size
echo 1
print
insert 5
find 6
breadth
size
size
print
balanced
size
echo 7
echo 0
find 4
echo 1
breadth
print
print
size
distance
size
distance
find 5
breadth
echo 0
breadth
find 2
find 5
insert 4
breadth
insert 1
print
balanced
breadth
distance
echo 4
echo 0
breadth
balanced
find 3
balanced
insert 0
breadth
balanced
find 6
echo 4
distance
size
print
balanced
find 2
find 7
distance
balanced
find 5
print
find 3
distance
size
balanced
insert 5
insert 7
find 7
print
find 7
size
find 6
size
breadth
distance
find 1
print
echo 6
balanced
balanced
balanced
breadth
balanced
print
balanced
balanced
breadth
distance
print
echo 4
breadth
breadth
size
find 5
balanced
balanced
size
balanced
print
print
size
breadth
balanced
balanced
echo 3
distance
find 5
distance
insert 6
balanced
find 1
balanced
find 5
distance
insert 0
print
distance
find 1
breadth
insert 2
insert 5
size
find 5
find 0
insert 5
distance
insert 4
insert 5
print
breadth
size
insert 1
size
distance
distance
breadth
distance
print
echo 2
balanced
size
distance
size
echo 0
balanced
print
distance
size
insert 1
insert 0
balanced
distance
distance
balanced
size
insert 6
find 4
echo 7
balanced
print
distance
echo 0
breadth
breadth
size
breadth
breadth
balanced
balanced
size
breadth
distance
find 0
distance
distance
insert 3
find 4
insert 2
print
balanced
breadth
echo 5
insert 2
breadth
distance
insert 0
find 3
find 7
breadth
insert 2
breadth
echo 2
print
insert 1
echo 1
echo 1
distance
balanced
echo 7
balanced
print
insert 0
echo 0
distance
distance
insert 7
size
print
find 6
balanced
distance
echo 7
echo 4
find 7
print
distance
find 0
echo 7
find 3
echo 7
distance
breadth
print
balanced
balanced
find 6
distance
size
find 6
distance
echo 7
balanced
find 3
distance
balanced
echo 6
echo 5
distance
balanced
print
find 6
print
size
balanced
find 5
print
insert 3
distance
print
find 4
find 3
echo 3
insert 3
echo 6
find 1
balanced
find 7
echo 3
insert 5
find 0
breadth
echo 7
find 7
find 7
find 7
size
size
size
insert 3
print
distance
print
balanced
distance
print
distance
insert 2
insert 6
breadth
print
find 1
balanced
insert 5
echo 1
find 3
echo 5
insert 1
size
echo 0
insert 1
size
balanced
distance
echo 0
print
print
size
print
breadth
balanced
size
size
find 3
find 5
distance
echo 5
print
print
balanced
distance
echo 4
find 2
distance
distance
breadth
breadth
print
size
print
find 5
size
find 3
find 7
insert 7
find 2
distance
size
breadth
size
print
insert 7
distance
insert 6
breadth
breadth
echo 0
print
balanced
size
insert 0
echo 5
echo 0
size
size
insert 4
find 7
echo 3
insert 1
insert 1
balanced
balanced
distance
size
size
distance
insert 3
find 6
echo 1
distance
insert 2
insert 0
find 4
breadth
find 4
size
breadth
insert 1
print
insert 3
balanced
size
breadth
insert 5
find 5
balanced
echo 4
insert 1
find 3
balanced
breadth
insert 1
find 2
breadth
distance
distance
breadth
distance
breadth
balanced
balanced
breadth
echo 7
size
print
insert 1
find 1
find 2
breadth
breadth
balanced
echo 3
find 5
balanced
distance
insert 7
distance
breadth
print
insert 7
insert 1
size
breadth
size
balanced
breadth
find 6
find 5
print
balanced
size
balanced
balanced
distance
print
breadth
insert 5
insert 7
breadth
echo 3
size
find 5
find 5
print
insert 7
insert 3
print
find 0
breadth
echo 1
echo 1
distance
print
echo 7
size
distance
distance
size
echo 1
echo 6
insert 1
breadth
size
size
size
echo 2
balanced
print
breadth
breadth
distance
find 4
breadth
distance
find 5
balanced
echo 5
find 4
print
insert 5
balanced
insert 1
balanced
insert 5
size
find 7
find 0
breadth
print
distance
insert 7
print
insert 1
find 7
print
print
breadth
insert 5
size
echo 3
size
breadth
size
breadth
breadth
breadth
echo 7
echo 2
insert 1
balanced
echo 6
insert 3
echo 6
size
distance
balanced
distance
breadth
balanced
distance
distance
print
distance
echo 4
echo 7
breadth
size
balanced
size
print
print
echo 4
insert 3
find 5
balanced
breadth
distance
print
distance
find 4
find 6
balanced
echo 5
balanced
echo 2
size
balanced
balanced
balanced
insert 3
breadth
print
echo 3
balanced
print
print
find 7
print
size
breadth
distance
balanced
insert 6
distance
insert 0
balanced
distance
insert 3
find 6
print
find 7
distance
print
size
breadth
insert 7
balanced
insert 0
balanced
insert 0
balanced
balanced
find 5
find 0
insert 6
breadth
echo 4
size
breadth
find 4
print
breadth
breadth
find 7
balanced
breadth
print
print
print
breadth
balanced
insert 2
breadth
find 4
echo 2
print
print
balanced
insert 1
distance
balanced
distance
balanced
find 2
size
insert 1
find 1
echo 2
distance
find 3
size
find 0
size
echo 3
distance
find 6
insert 5
insert 1
print
find 5
distance
balanced
balanced
size
size
print
distance
find 0
find 0
find 3
insert 1
distance
balanced
size
breadth
insert 6
size
print
balanced
balanced
breadth
balanced
find 3
find 2
distance
balanced
print
size
distance
balanced
size
distance
insert 5
distance
insert 0
size
insert 6
insert 1
print
find 4
size
insert 2
echo 6
size
echo 6
insert 1
insert 2
balanced
size
find 6
find 3
size
balanced
insert 3
echo 3
breadth
distance
insert 6
size
breadth
print
print
distance
echo 4
size
print
echo 5
size
insert 7
size
balanced
print
distance
print
echo 0
size
insert 4
insert 6
size
echo 1
print
size
size
print
balanced
breadth
distance
find 1
distance
print
size
size
breadth
size
breadth
print
distance
distance
size
print
size
balanced
size
breadth
size
distance
echo 4
balanced
find 5
print
print
size
distance
find 0
find 0
insert 7
echo 5
distance
find 2
size
print
size
echo 4
insert 1
find 4
print
distance
find 6
balanced
breadth
distance
echo 7
balanced
insert 6
insert 2
echo 3
print
size
find 4
echo 3
insert 7
balanced
find 0
breadth
echo 2
breadth
insert 6
breadth
find 1
find 6
insert 4
insert 3
breadth
size
balanced
echo 2
balanced
echo 7
insert 3
breadth
echo 1
echo 2
balanced
size
find 2
distance
insert 4
size
find 7
echo 4
distance
find 0
balanced
insert 3
insert 0
insert 6
insert 1
size
distance
find 5
find 1
distance
insert 4
find 0
print
balanced
