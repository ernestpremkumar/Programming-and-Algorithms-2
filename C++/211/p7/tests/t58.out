2
{}
<2> is not in tree.
{0}
{0}
{0}
1
Average distance of nodes to root = 0
{0}
Average distance of nodes to root = 0
{0}
{0}
{0}
<3> is not in tree.
Average distance of nodes to root = 0
Average distance of nodes to root = 0
{0}
{0}
{0}
0
2
{0, 4}
Average distance of nodes to root = 0.5
<1> is not in tree.
2
<0> is in tree.
<6> is not in tree.
2
6
<7> is not in tree.
7
{0, 4}
Tree is balanced.
Tree is balanced.
2
2
1
{0, 4, 6}
{0, 4, 6}
Tree is not balanced.
Average distance of nodes to root = 1
<0> is in tree.
Tree is not balanced.
Average distance of nodes to root = 1
Average distance of nodes to root = 1
{0, 4, 6}
3
{0, 4, 6}
{0, 4, 6}
<0> is in tree.
<3> is not in tree.
{0, 4, 6}
Tree is not balanced.
6
Average distance of nodes to root = 1
{0, 4, 6}
<5> is not in tree.
{0, 4, 6}
1
Average distance of nodes to root = 1
Average distance of nodes to root = 1
Average distance of nodes to root = 1
{0, 4, 6}
3
{0, 4, 6}
<5> is not in tree.
{0, 4, 6}
2
<2> is not in tree.
Average distance of nodes to root = 1.5
{0, 4, 6, 5}
{0, 4, 6, 5}
4
{0, 4, 5, 6}
Tree is not balanced.
{0, 4, 6, 5}
Average distance of nodes to root = 1.5
Tree is not balanced.
Average distance of nodes to root = 1.5
Average distance of nodes to root = 1.5
{0, 4, 6, 5}
{0, 4, 5, 6}
5
{0, 4, 5, 6}
Average distance of nodes to root = 1.5
1
<3> is not in tree.
2
<7> is not in tree.
4
{0, 4, 5, 6}
Tree is not balanced.
{0, 3, 4, 5, 6}
5
5
5
0
<1> is not in tree.
<3> is in tree.
<3> is in tree.
5
Tree is not balanced.
{0, 4, 3, 6, 5}
5
7
Tree is not balanced.
Average distance of nodes to root = 1.6
Average distance of nodes to root = 1.6
0
<0> is in tree.
Tree is not balanced.
<4> is in tree.
{0, 4, 3, 6, 5}
5
5
Tree is not balanced.
5
{0, 3, 4, 5, 6}
0
Tree is not balanced.
0
5
Average distance of nodes to root = 1.6
{0, 1, 3, 4, 5, 6}
{0, 4, 3, 6, 1, 5}
Tree is not balanced.
Tree is not balanced.
<1> is in tree.
Tree is not balanced.
Average distance of nodes to root = 1.83333
7
Average distance of nodes to root = 2.14286
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6}
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 2}
{0, 4, 3, 6, 1, 5, 2}
Tree is not balanced.
Average distance of nodes to root = 2.14286
Average distance of nodes to root = 2.14286
{0, 1, 2, 3, 4, 5, 6}
Average distance of nodes to root = 2.14286
Tree is not balanced.
Tree is not balanced.
0
{0, 4, 3, 6, 1, 5, 2}
Average distance of nodes to root = 2.14286
{0, 4, 3, 6, 1, 5, 2}
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 2}
Tree is not balanced.
7
{0, 1, 2, 3, 4, 5, 6}
{0, 1, 2, 3, 4, 5, 6}
<0> is in tree.
4
{0, 4, 3, 6, 1, 5, 2}
7
{0, 4, 3, 6, 1, 5, 2}
7
<4> is in tree.
4
{0, 1, 2, 3, 4, 5, 6}
7
{0, 1, 2, 3, 4, 5, 6}
{0, 4, 3, 6, 1, 5, 2}
7
Average distance of nodes to root = 2.14286
{0, 4, 3, 6, 1, 5, 2}
6
3
Average distance of nodes to root = 2.14286
{0, 1, 2, 3, 4, 5, 6}
3
Tree is not balanced.
7
1
{0, 1, 2, 3, 4, 5, 6}
<6> is in tree.
{0, 4, 3, 6, 1, 5, 2}
7
7
{0, 1, 2, 3, 4, 5, 6}
Tree is not balanced.
7
7
0
<4> is in tree.
1
{0, 4, 3, 6, 1, 5, 2}
{0, 1, 2, 3, 4, 5, 6}
{0, 1, 2, 3, 4, 5, 6}
7
Average distance of nodes to root = 2.14286
7
Average distance of nodes to root = 2.14286
<5> is in tree.
{0, 4, 3, 6, 1, 5, 2}
0
{0, 4, 3, 6, 1, 5, 2}
<2> is in tree.
<5> is in tree.
{0, 4, 3, 6, 1, 5, 2}
{0, 1, 2, 3, 4, 5, 6}
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 2}
Average distance of nodes to root = 2.14286
4
0
{0, 4, 3, 6, 1, 5, 2}
Tree is not balanced.
<3> is in tree.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 2}
Tree is not balanced.
<6> is in tree.
4
Average distance of nodes to root = 2.14286
7
{0, 1, 2, 3, 4, 5, 6}
Tree is not balanced.
<2> is in tree.
<7> is not in tree.
Average distance of nodes to root = 2.14286
Tree is not balanced.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6}
<3> is in tree.
Average distance of nodes to root = 2.14286
7
Tree is not balanced.
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
8
<6> is in tree.
8
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
6
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
4
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
8
<5> is in tree.
Tree is not balanced.
Tree is not balanced.
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
Tree is not balanced.
3
Average distance of nodes to root = 2.25
<5> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
<1> is in tree.
Tree is not balanced.
<5> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<1> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
8
<5> is in tree.
<0> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 4, 3, 6, 1, 5, 7, 2}
8
8
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
2
Tree is not balanced.
8
Average distance of nodes to root = 2.25
8
0
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
8
Tree is not balanced.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Tree is not balanced.
8
<4> is in tree.
7
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
0
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
8
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
Tree is not balanced.
8
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
<0> is in tree.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
5
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
<3> is in tree.
<7> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
2
{0, 1, 2, 3, 4, 5, 6, 7}
1
1
Average distance of nodes to root = 2.25
Tree is not balanced.
7
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
0
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
8
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.25
7
4
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<0> is in tree.
7
<3> is in tree.
7
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
<6> is in tree.
Average distance of nodes to root = 2.25
8
<6> is in tree.
Average distance of nodes to root = 2.25
7
Tree is not balanced.
<3> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
6
5
Average distance of nodes to root = 2.25
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
<3> is in tree.
3
6
<1> is in tree.
Tree is not balanced.
<7> is in tree.
3
<0> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
7
<7> is in tree.
<7> is in tree.
<7> is in tree.
8
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
Tree is not balanced.
1
<3> is in tree.
5
8
0
8
Tree is not balanced.
Average distance of nodes to root = 2.25
0
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
8
8
<3> is in tree.
<5> is in tree.
Average distance of nodes to root = 2.25
5
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.25
4
<2> is in tree.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
8
<3> is in tree.
<7> is in tree.
<2> is in tree.
Average distance of nodes to root = 2.25
8
{0, 4, 3, 6, 1, 5, 7, 2}
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
0
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
5
0
8
8
<7> is in tree.
3
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.25
8
8
Average distance of nodes to root = 2.25
<6> is in tree.
1
Average distance of nodes to root = 2.25
<4> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
<4> is in tree.
8
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
{0, 4, 3, 6, 1, 5, 7, 2}
<5> is in tree.
Tree is not balanced.
4
<3> is in tree.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
<2> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
7
8
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
<2> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
3
<5> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 4, 3, 6, 1, 5, 7, 2}
8
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
<6> is in tree.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
3
8
<5> is in tree.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
1
1
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
7
8
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
8
1
6
{0, 4, 3, 6, 1, 5, 7, 2}
8
8
8
2
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
<4> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
<5> is in tree.
Tree is not balanced.
5
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
8
<7> is in tree.
<0> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 4, 3, 6, 1, 5, 7, 2}
8
3
8
{0, 4, 3, 6, 1, 5, 7, 2}
8
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
7
2
Tree is not balanced.
6
6
8
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
4
7
{0, 4, 3, 6, 1, 5, 7, 2}
8
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
4
<5> is in tree.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<4> is in tree.
<6> is in tree.
Tree is not balanced.
5
Tree is not balanced.
2
8
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
3
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
<5> is in tree.
<0> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
4
8
{0, 4, 3, 6, 1, 5, 7, 2}
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
<7> is in tree.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
<4> is in tree.
2
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
Tree is not balanced.
<2> is in tree.
8
<1> is in tree.
2
Average distance of nodes to root = 2.25
<3> is in tree.
8
<0> is in tree.
8
3
Average distance of nodes to root = 2.25
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
Tree is not balanced.
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<0> is in tree.
<0> is in tree.
<3> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
8
{0, 4, 3, 6, 1, 5, 7, 2}
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
Tree is not balanced.
<3> is in tree.
<2> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 2.25
Tree is not balanced.
8
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
8
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
8
6
8
6
Tree is not balanced.
8
<6> is in tree.
<3> is in tree.
8
Tree is not balanced.
3
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
8
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
4
8
{0, 1, 2, 3, 4, 5, 6, 7}
5
8
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
0
8
8
1
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
<1> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
{0, 4, 3, 6, 1, 5, 7, 2}
8
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
8
{0, 4, 3, 6, 1, 5, 7, 2}
8
Average distance of nodes to root = 2.25
4
Tree is not balanced.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 2.25
<0> is in tree.
<0> is in tree.
5
Average distance of nodes to root = 2.25
<2> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
4
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<6> is in tree.
Tree is not balanced.
{0, 4, 3, 6, 1, 5, 7, 2}
Average distance of nodes to root = 2.25
7
Tree is not balanced.
3
{0, 1, 2, 3, 4, 5, 6, 7}
8
<4> is in tree.
3
Tree is not balanced.
<0> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
2
{0, 4, 3, 6, 1, 5, 7, 2}
{0, 4, 3, 6, 1, 5, 7, 2}
<1> is in tree.
<6> is in tree.
{0, 4, 3, 6, 1, 5, 7, 2}
8
Tree is not balanced.
2
Tree is not balanced.
7
{0, 4, 3, 6, 1, 5, 7, 2}
1
2
Tree is not balanced.
8
<2> is in tree.
Average distance of nodes to root = 2.25
8
<7> is in tree.
4
Average distance of nodes to root = 2.25
<0> is in tree.
Tree is not balanced.
8
Average distance of nodes to root = 2.25
<5> is in tree.
<1> is in tree.
Average distance of nodes to root = 2.25
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
